### DISCUSSION TITLE
Fan speed showing as lightbulb

### DISCUSSION THREAD

**Discussion by Many-sun** [merdok/homebridge-miot#344] [Q&A] [answered]
Hi, I'm new to homebridge, so maybe it's a silly question. 

I have a MI Air Purifier 3C and I want to have the "Favorite speed" as a propertyControl. It was very simple to use, but now it's showing as a lightbulb instead of a fan or something. Is there anyway I can change this please?

Device model is zhimi.airpurifier.mb4

And the property is set as simply as

"propertyControl": [
                        {
                            "property": "custom-service:favorite-speed",
                            "name": "Favourite speed",
                            "type": "unit16"
                        }

**Comment by merdok** [comment, marked as answer]
You are right, right now the default way for showing such properties is a lightbulb, hence when the plugin detects that the property is a value range it will create a lightbulb. I might make it configurable at some point, but there is no way right now to configure that.
Could you please open an issue with a feature request for that, so that it doe snot get lost?

**Reply by Many-sun** [reply]
Thanks for answering, I'll make a feature request :)

**Reply by Many-sun** [reply]
For any future reference this is the feature-request https://github.com/merdok/homebridge-miot/issues/345. I hope the description is ok.
